tests/lib/schema_validation_test.py

```diff
@@ -143,10 +143,52 @@ def test_valid_without_auto_neg_and_missing_speed(self, default_data):
         not a valid configuration, however, this is not handled by the schema
         at the moment, deferring the handling to the application code.
         """
-        default_data[INTERFACES][0].update(
+        default_data[Interface.KEY][0].update(
             {'type': 'ethernet', 'auto-negotiation': False}
         )
-        del default_data[INTERFACES][0]['link-speed']
+        del default_data[Interface.KEY][0]['link-speed']
+
+        libnmstate.validator.validate(default_data)
+
+
+class TestIfaceTypeVxlan(object):
+    def test_invalid_vxlan_with_bad_id_type(self, default_data):
```

Comment:
I would recommend to drop the `vxlan` name from the test name. The class name already has this and provides the needed context.

Also, for a smoother reading of the test name, you can leave the expected outcome at the end:
`test_bad_ip_type_is_invalid`